Core/Source/MVVM/ViewModel.swift

```diff
@@ -41,44 +41,58 @@ public protocol ViewModel {
 /// Wraps an `Action` with additional data to be rendered in a "secondary" context like context menus or long-press
 /// menus.
 public struct SecondaryActionInfo {
-
-    public init(action: Action, title: String, state: State = .off, enabled: Bool = true, imageName: String? = nil) {
-        self.action = action
-        self.title = title
-        self.state = state
-        self.enabled = enabled
-        self.imageName = imageName
+    public struct Metadata {
+        public let title: String
+        public let state: State
+        public let enabled: Bool
+        public let imageName: String?
+        public let keyEquivalent: String
+
+        /// State of the secondary action. Note that this differs from enabled, but instead represents whether the
+        /// action is "checked" in a list.
+        public enum State {
+            case on
+            case off
+            case mixed
+        }
     }
 
-    /// State of the secondary action. Note that this differs from enabled, but instead represents whether the action
-    /// is "checked" in a list.
-    public enum State {
-        case on
-        case off
-        case mixed
+    public init(
```

Comment:
Since we're exposing the metadata property publicly from these structs why don't we just take a `Metadata` in the init to avoid having to change both when the fields change.

`public init(action:metadata:)`

Reply:
Makes sense. I was using the different signatures for .action and .nested to enforce some conventions (such as nested menus don't have keyboard shortcuts, as far as I can tell) but I can use convenience inits on the Metadata for that.

Reply:
Oh, and the other reason is this would break like 60 call sites in $MAIN_APP_WERE_WORKING_ON :-/

Reply:
Looked into fixing these but it got ugly quick, especially for the common action+tile case:
`SecondaryActionInfo(title: "Open".localized, action: open)`
vs
`SecondaryActionInfo(metadata: SecondaryActionInfo.Metadata(title: "Open".localized), action: open)`

Reply:
Lets fix that in our app with a convenience extension in app code rather than crimping the API here.

Reply:
^ Will support in extension to Pilot in app code.